src/compression.rs

```diff
@@ -557,3 +586,251 @@ pub fn compress_graph<K: Kmer, D: Clone + Debug, S: CompressionSpec<D>>(
 ) -> DebruijnGraph<K, D> {
     CompressFromGraph::<K, D, S>::compress_graph(stranded, spec, old_graph, censor_nodes)
 }
+
+
+//////////////////////////////
+// Compress from Hash a new Struct
+//////////////////////////////
+/// Generate a compressed DeBruijn graph from hash_index
+struct CompressFromHash<'a, K: 'a + Kmer, D: 'a, S: CompressionSpec<D>> {
+    stranded: bool,
+    k: PhantomData<K>,
+    d: PhantomData<D>,
+    spec: S,
+    available_kmers: BitSet,
+    index: &'a boomphf::BoomHashMap2<K, Exts, D>,
```

Comment:
We can merge CompressFromKmers and CompressFromHash into the same implementation with the following trick: make the index field a generic type that implement a trait like this: 
```
trait KmerIndex {
    fn get_kmer_data(&'a self, kmer: &K) -> (&Exts, &'a D);
    fn get_kmer_id(&self, kmer: &K) -> Option<usize>;
}
```
then create impls of this trait for Vec<(K,Exts,D)> and BookHashMap2<K, Exts, D>.  
Then you can reuse almost all the code between CompressFromHash and CompressFromKmer.

Reply:
not sure if it's worth doing this... maybe not. We may just want to get rid of the Vec<> version.